src/source-control/GitSourceControl.ts

```diff
@@ -263,64 +265,82 @@ export class GitSourceControl implements SourceControl {
-        const childrenOfTargetCommit = draftTargetCommitSttack.childCommits;
+        const draftTargetCommit = draftRepo.commits.get(targetCommitHash)!;
+        const childrenOfTargetCommit = draftTargetCommit.childCommits;
+
+        // Add the newly-created commit to our repo
         const newCommitToInsert: Commit = {
           hash: newCommitOid.tostrS(),
-          title: baseCommitSttack.title,
+          title: commitToBeRebased.title,
           timestamp: new Date(),
-          author: baseCommitSttack.author,
-          committer: baseCommitSttack.committer,
-          refNames: baseCommitSttack.refNames.filter(refIsLocal),
-          parentCommits: [draftTargetCommitSttack.hash],
+          author: commitToBeRebased.author,
+          committer: commitToBeRebased.committer,
+          refNames: commitToBeRebased.refNames.filter(refIsLocal),
+          parentCommits: [draftTargetCommit.hash],
           childCommits: [...childrenOfTargetCommit],
         };
         draftRepo.commits.set(newCommitOid.tostrS(), newCommitToInsert);
 
-        // Update ChildCommitHash for targetCommitSttack to newCommit
-        draftTargetCommitSttack.childCommits = [newCommitOid.tostrS()];
-        // Update ParentCommitHash for all of the children and replace targetCommitSttack's hash with newCommit's hash
-        childrenOfTargetCommit.forEach((child) => {
-          const childCommit = draftRepo.commits.get(child)!;
-          const parentsOfChild = childCommit.parentCommits.map((parent) => {
-            if (parent === draftTargetCommitSttack.hash) {
-              return newCommitOid.tostrS();
-            }
-            return parent;
-          });
-          childCommit.parentCommits = parentsOfChild;
-          draftRepo.commits.set(child, childCommit);
+        // Update target's children
+        draftTargetCommit.childCommits = [newCommitOid.tostrS()];
+
+        // Update ParentCommitHash for all of the children and replace targetCommit's hash with newCommit's hash
+        childrenOfTargetCommit.forEach((childHash) => {
+          const childCommit = draftRepo.commits.get(childHash)!;
+          const newParentCommitHashes = childCommit.parentCommits.map(
+            (parent) => {
+              // Replace pre-cherry-pick parent with the post-cherry-pick one
+              if (parent === draftTargetCommit.hash) {
+                return newCommitOid.tostrS();
+              }
+              return parent;
+            },
+          );
+          childCommit.parentCommits = newParentCommitHashes;
+          draftRepo.commits.set(childHash, childCommit);
         });
       });
+
+      queue.push(...commitToBeRebased.childCommits);
+
       // We now would want to use this new commit hash as the target
-      targetCommit = newCommitOid.tostrS();
+      // FIXME: This will turn a commit tree into a linear stack!
+      targetCommitHash = newCommitOid.tostrS();
```

Comment:
👍 - We can handle this later I guess, what we would need is a level order traversal rather than a BFS. Wouldn't that work?